common-arcana.lic

```diff
@@ -31,12 +31,16 @@ module DRCA
     end
   end
 
-  def harness_mana(amount)
-    amount.each do |mana|
-      result = DRC.bput("harness #{mana}", 'You tap into', 'Strain though you may')
-      pause 0.5
-      waitrt?
-      break if result =~ /Strain though you may/
+  def harness?(mana)
+    result = DRC.bput("harness #{mana}", 'You tap into', 'Strain though you may')
+    pause 0.5
+    waitrt?
+    return result =~ /You tap into/
+  end
+
+  def harness_mana(amounts)
+    amounts.each do |mana|
+      break unless harness?(mana)
     end
   end
```

Comment:
Moved the logic for harnessing from the loop in `harness_mana` into its own method `harness?`. The new `harness?` predicate is used in the `charge?` method if you're too injured to charge cambrinth.

Reply:
:warning: `combat-trainer` calls `def harness_mana` at line 3816 - `harness_mana(@charges.flatten)` in the `def check_harness` function.

Reply:
Thanks for checking on the usages. Could you elaborate on your concerns?

The `harness_mana` method is not going away and its behavior is still the same per my regression test “Harness multiple amounts” (bottom of the PR description).

Reply:
I just wanted to make you aware of the scripts using this function. 

Have you tested with runestone casting?

Reply:
> Have you tested with runestone casting?

I haven't, no. Is that something you could help with?

Reply:
> > Have you tested with runestone casting?
> 
> I haven't, no. Is that something you could help with?

I just copied and pasted the contents into my `common-arcana` file. I will run tomorrow, Friday, and make sure all is working as expected.

Reply:
Thank you!

Reply:
Used all day and seems fine. I have not seen anything unusual.